### PR TITLE
Adding an exhaustive list of Helm chart values to eck-elasticsearch chart.

### PR DESCRIPTION
**Pull request by naemono**
This change adds an exhaustive list of Helm chart values to the eck-elasticsearch Helm chart.

This expands the `podTemplate` field in the values file and includes all *relevant* options within the `podTemplate` field.

relates to #6187

### CODE REVIEW

File: deploy/eck-elasticsearch/values.yaml [hunk 1]
```diff
@@ -32,31 +32,79 @@ annotations: {}
...
+  #   - key: gcs_client_2
+  #     path: gcs.client.client_2.credentials_file
 
 # Settings for limiting the number of simultaneous changes to an Elasticsearch resource.
 # ref: https://www.elastic.co/guide/en/cloud-on-k8s/current/k8s-update-strategy.html
 #
 updateStrategy: {}
+  # changeBudget:
+  #     maxSurge: 3
+  #     maxUnavailable: 1
```
Comment (pebrc) [inline deploy/eck-elasticsearch/values.yaml:98]: Indentation feels off here.

File: deploy/eck-elasticsearch/values.yaml [hunk 2]
```diff
@@ -81,10 +129,355 @@ nodeSets:
...
+        # An optional identifier to prepend to each key in the ConfigMap.
+          # prefix: "DS_"
+        # Set environment variables from key/value pairs within a Secret
+        # - secretRef:
+        #     name: specialEnvSecretConfig
+            # Specify whether the Secret must be defined
+            # optional: false
+
+        # Container image name.
+        # image: ""
```
Comment (pebrc) [inline deploy/eck-elasticsearch/values.yaml:223]: Same here: while it is technically possible to set the image value in the PodTemplate directly, I would not advertise this as we have a top level field in the CRD that should be used for this.

File: deploy/eck-elasticsearch/values.yaml [hunk 3]
```diff
@@ -81,10 +129,355 @@ nodeSets:
...
+    # Scheduler will schedule pods in a way which abides by the constraints. All topologySpreadConstraints are ANDed.
+    #
+    # These settings are generally applied within each `nodeSets[].podTemplate` field to apply to a specific Elasticsearch nodeset.
+    #
+    # https://www.elastic.co/guide/en/cloud-on-k8s/current/k8s-advanced-node-scheduling.html
+    # topologySpreadConstraints: {}
+
+    # List of volumes that can be mounted by containers belonging to the pod.
+    # https://kubernetes.io/docs/concepts/storage/volumes
+    # volumes: []
```
Comment (pebrc) [inline deploy/eck-elasticsearch/values.yaml:483]: I have reservations against dumping the whole PodTemplateSpec including docs here. Would a link to the Kubernetes doc not suffice?

**Review by pebrc — COMMENTED** [review]
I am not sure about this one. I am not sure duplicating the PodTemplateSpec is the right approach. How do we keep that updated etc. Would it not suffice to link to the k8s API docs?

**Comment by naemono** [comment]
> I am not sure duplicating the PodTemplateSpec

How do you feel about me documenting the most commonly used/overridden options in the podtemplatespec?  While also pointing to the documentation?

File: deploy/eck-elasticsearch/values.yaml [hunk 4]
```diff
@@ -32,31 +32,79 @@ annotations: {}
...
+  #     labels:
+  #       my-custom: label
+  #   spec:
+  #     type: LoadBalancer
+  # tls:
+  #   subjectAltNames:
+  #     - ip: 1.2.3.4
+  #     - dns: hulk.example.com
+  #   certificate:
+  #     secretName: custom-ca
```
Comment (thbkrkr) [inline deploy/eck-elasticsearch/values.yaml:76]: Compared to the `transport` service, the `http` service expects the `selfSignedCertificate` field above `tls`.
Also, do we want to show how to disable it?
```suggestion
  # tls:
  #   selfSignedCertificate:
  #     disabled: false
  #     subjectAltNames:
  #       - ip: 1.2.3.4
  #       - dns: hulk.example.com
  #   certificate:
  #     secretName: custom-ca
```

File: deploy/eck-elasticsearch/values.yaml (hunk 1 above)
Comment (naemono) [reply deploy/eck-elasticsearch/values.yaml:102]: If I select the commented out sections, and use my editor to remove all comments, it becomes perfectly indented.  My intent here was to make it easy to remove the comments using some editor, and have them aligned correctly.

Were you wanting to see them more like the following?
```
updateStrategy: {}
# changeBudget:
#     maxSurge: 3
#     maxUnavailable: 1
```

**Comment by naemono** [comment]
@pebrc I've trimmed down the number of fields shown in the `podTemplateSpec` section.  Let me know what you think.  Thanks.

File: deploy/eck-elasticsearch/values.yaml (hunk 1 above)
Comment (thbkrkr) [inline deploy/eck-elasticsearch/values.yaml:102]: I also see too much indentation for the `maxXyz` fields:
```suggestion
  # changeBudget:
  #   maxSurge: 3
  #   maxUnavailable: 1
```

File: deploy/eck-elasticsearch/Chart.yaml [hunk 5]
```diff
@@ -3,7 +3,7 @@ name: eck-elasticsearch
 description: A Helm chart to deploy Elasticsearch managed by the ECK Operator.
 kubeVersion: ">= 1.21.0-0"
 type: application
-version: 0.2.0
+version: 0.2.1
```
Comment (thbkrkr) [inline deploy/eck-elasticsearch/Chart.yaml:6]: I think the version in the Helm tests needs to be updated as well.

File: deploy/eck-elasticsearch/values.yaml (hunk 1 above)
Comment (pebrc) [reply deploy/eck-elasticsearch/values.yaml:107]: Sorry maybe it is just the diff UI here in GH as it looks like it is indented by 4 spaces instead of 2.

File: deploy/eck-elasticsearch/values.yaml [hunk 6]
```diff
@@ -85,11 +137,208 @@ nodeSets:
...
+    #           - S1
+    #       topologyKey: topology.kubernetes.io/zone
+    #   podAntiAffinity:
+    #     preferredDuringSchedulingIgnoredDuringExecution:
+    #     - weight: 100
+    #       podAffinityTerm:
+    #         labelSelector:
+    #           matchLabels:
+    #             elasticsearch.k8s.elastic.co/cluster-name: quickstart
+    #         topologyKey: kubernetes.io/hostname
```
Comment (pebrc) [inline deploy/eck-elasticsearch/values.yaml:185]: I would probably shorten this to use just one example, shorter examples are easier to read and understand.

File: deploy/eck-elasticsearch/values.yaml [hunk 7]
```diff
@@ -85,11 +137,208 @@ nodeSets:
...
+        #   fieldRef:
+        #     fieldPath: spec.nodeName
+        #   Set the value from the container's resource requests.
+        #   resourceFieldRef:
+        #     containerName: "elasticsearch"
+        #     resource: requests.cpu
+        #   Set the value from a secret.
+        #   secretKeyRef:
+        #     name: specialEnvSecretConfig
+        #     key: "my-key"
```
Comment (pebrc) [inline deploy/eck-elasticsearch/values.yaml:212]: Same here with the env vars (above and below my comment)  I would use a shorter example and maybe a link to the K8s docs.

**Review by pebrc — APPROVED** [review]
LGTM, I think we  really have to fix the Helm versioning. I just bumped the charts for a change that introduced the version to the next minor and here we are doing it again but to a patch version. @thbkrkr you have to probably consolidate these version bumps before creating the next release branch.

File: deploy/eck-elasticsearch/values.yaml (hunk 1 above)
Comment (naemono) [reply deploy/eck-elasticsearch/values.yaml]: @thbkrkr pointed out what I was not seeing.  I thought y'all were referring to the comment's indentation, not the `maxSurge, maxUnavailable` fields.  I've fixed this.

File: deploy/eck-elasticsearch/values.yaml [hunk 8]
```diff
@@ -85,11 +137,151 @@ nodeSets:
...
+    #           - antarctica-west1
+
+    # Containers array.  Should only be used to customize the 'elasticsearch' container using the following fields.
       containers:
       - name: elasticsearch
+
+        # List of environment variables to set in the 'elasticsearch' container.
+        # https://kubernetes.io/docs/tasks/inject-data-application/define-environment-variable-container/
+        # env:
+        # name/value environment variable pair to set within the 'elasticsearch' container.
```
Comment (thbkrkr) [inline deploy/eck-elasticsearch/values.yaml:170]: Not sure we need this:
```suggestion
```

File: deploy/eck-elasticsearch/values.yaml [hunk 9]
```diff
@@ -85,11 +137,151 @@ nodeSets:
...
+          # that it does not run as UID 0 (root) and fail to start the container if it does. If unset or false, no such validation will be performed.
+          # May also be set in PodSecurityContext. If set in both SecurityContext and PodSecurityContext, the value specified in SecurityContext takes precedence.
+          # runAsNonRoot: true
+
+          # The UID to run the entrypoint of the container process. Defaults to user specified in image metadata if unspecified. May also be set in PodSecurityContext.
+          # If set in both SecurityContext and PodSecurityContext, the value specified in SecurityContext takes precedence.
+          # runAsUser: 1000
+
+    # ImagePullSecrets is an optional list of references to secrets in the same namespace to use for pulling any of the images used by this PodSpec.
+    # If specified, these secrets will be passed to individual puller implementations for them to use.
```
Comment (thbkrkr) [inline deploy/eck-elasticsearch/values.yaml:220]: May be too detailed:
```suggestion
```

File: deploy/eck-elasticsearch/values.yaml [hunk 10]
```diff
@@ -85,11 +137,151 @@ nodeSets:
...
+
+    # List of initialization containers belonging to the pod. Init containers are executed in order prior to containers being started.
+    # If any init container fails, the pod is considered to have failed and is handled according to its restartPolicy.
+    # The name for an init container or normal container must be unique among all containers.
+    # Init containers may not have Lifecycle actions, Readiness probes, Liveness probes, or Startup probes.
+    # The resourceRequirements of an init container are taken into account during scheduling
+    # by finding the highest request/limit for each resource type, and then using the max of of that value or the sum of the normal containers.
+    # Limits are applied to init containers in a similar fashion. Init containers cannot currently be added or removed. Cannot be updated.
+    #
+    # See previous 'containers' field for a common list of fields that apply.
```
Comment (thbkrkr) [inline deploy/eck-elasticsearch/values.yaml:233]: May be too detailed:
```suggestion
    List of initialization containers belonging to the pod.
```

File: deploy/eck-elasticsearch/values.yaml [hunk 11]
```diff
@@ -85,11 +137,160 @@ nodeSets:
...
+            # cpu: 1
+            memory: 2Gi
+
+          # Example increasing both the requests and limits values:
+          # limits:
+          #   cpu: 4
+          #   memory: 8Gi
+          # requests:
+          #   cpu: 1
+          #   memory: 4Gi
```
Comment (thbkrkr) [inline deploy/eck-elasticsearch/values.yaml:193]:
```suggestion
          #   memory: 8Gi
```
I'd follow to the best practice memory limit=request.

File: deploy/eck-elasticsearch/values.yaml [hunk 12]
```diff
@@ -85,11 +137,160 @@ nodeSets:
...
+        # These typically are set automatically by the ECK Operator, and should only be adjusted
+        # with the full knowledge of the effects of each field.
+        #
+        # securityContext:
+
+          # Whether this container has a read-only root filesystem. Default is false.
+          # readOnlyRootFilesystem: false
+
+          # The GID to run the entrypoint of the container process. Uses runtime default if unset. May also be set in PodSecurityContext.
+          # If set in both SecurityContext and PodSecurityContext, the value specified in SecurityContext takes precedence.
```
Comment (thbkrkr) [inline deploy/eck-elasticsearch/values.yaml:207]:
```suggestion
```

**Review by thbkrkr — APPROVED** [review]
LGTM
